src/farmhashuo.rs

```diff
@@ -28,19 +28,19 @@ pub fn uo_hash64_with_seeds(mut s: &[u8], seed0: u64, seed1: u64) -> u64 {
     x = x.wrapping_mul(K2);
     let mul = K2.wrapping_add(u & 0x82);
 
-    // Set end so that after the loop we have 1 to 64 bytes left to process.
-    let end = &s[((len - 1) / 64) * 64..];
```

Comment:
Removing this helped (it's unused). Unfortunately, panic on out of bounds is a side effect that means an unused expression like this can not be optimized out :-(